### PR TITLE
Add vhd-tool to xs-extra

### PR DESCRIPTION
**Pull request by gaborigloi**
Fixes https://github.com/xapi-project/xs-opam/issues/84.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

**Comment by gaborigloi** [comment]
From the point of view of https://github.com/ocaml/opam-repository/pull/9206, I wonder why adding vhd-tool is important, and why vhd-tool 0.7.8 was useful in the main opam repo, as its `opam` file doesn't even have an install step, so people can only build it.

File: packages/xs-extra/vhd-tool.master/opam [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  [make]
+]
+depends: [
+  "ocamlfind"
+  "lwt" {>= "2.4.3"}
+  "cstruct" {>= "1.0.1"}
+  "vhd-format" {>= "0.7.0"}
+  "uuidm"
+  "cmdliner"
+  "nbd" {>= "1.0.1"}
```
Comment (mseri) [inline packages/xs-extra/vhd-tool.master/opam]: & < "3.0.0"

**Comment by gaborigloi** [comment]
Oops, I noticed some dependencies are missing from the `opam` file, as the original seems to be a bit outdated, I'll update it.

File: packages/xs-extra/vhd-tool.master/opam [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "ounit"
+  "uri"
+  "tar-format"
+  "sha"
+  "cohttp" {>="0.12.0"}
```
Comment (mseri) [inline packages/xs-extra/vhd-tool.master/opam]: An upper bound is needed here as well < "0.22.0": http://rgrinberg.com/posts/cohttp-breakage-ahead/

File: packages/xs-extra/vhd-tool.master/opam [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+]
+build: [
+  ["oasis" "setup"]
+  ["./configure" "--bindir=%{bin}%" "--etcdir=%{etc}%" "--libexecdir=%{bin}%"]
+  [make]
+]
+depends: [
+  "oasis" {build}
+  "ocamlfind" {build}
+  "lwt" {>= "2.4.3"}
```
Comment (mseri) [inline packages/xs-extra/vhd-tool.master/opam]: < "3.0.0"

File: packages/xs-extra/vhd-tool.master/opam [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "cstruct" {>= "1.0.1"}
```
Comment (mseri) [inline packages/xs-extra/vhd-tool.master/opam]: This needs an upper bound as well

**Comment by gaborigloi** [comment]
Looks like `vhd-tool` will only work with older ocaml versions:
```
[vhd-tool] /mnt/ already up-to-date
The following dependencies couldn't be met:
  - vhd-tool -> xapi-tapctl -> xapi-stdext >= 0.13.0 -> sexplib < 113.01.00
Your request can't be satisfied:
  - sexplib<113.01.00 is not available because your system doesn't comply with ocaml-version >= "4.02.1" & ocaml-version < "4.03".
```

**Comment by mseri** [comment]
I am surprised, it did work on `opam-repository`

**Comment by gaborigloi** [comment]
Do we need the `sparse_dd` executable too? That depends on some optional xenstore stuff. If so, I'll add those too to the opam dependencies. Currently it would not compile `sparse_dd` because those optional dependencies are not listed.

**Comment by mseri** [comment]
Let's add them for now. (Just seen the edit: good!)

I think we might split he package in two when we port it to jbuilder

**Comment by mseri** [comment]
The installation is still failing due to some issues with the manpage :-(

**Comment by gaborigloi** [comment]
```
opam@ef21fc9801a6:~/.opam/system/build/vhd-tool.~unknown$ ls
CHANGES      _oasis                main.native      sparse_dd.native
LICENSE      _tags                 myocamlbuild.ml  src
MAINTAINERS  config.mk             opam             vhd-tool
Makefile     configure             setup.data       vhd-tool.1
README.md    configure.ml          setup.log        vhd-tool.install
_build       get_vhd_vsize.native  setup.ml
opam@ef21fc9801a6:~/.opam/system/build/vhd-tool.~unknown$ cat vhd-tool.install 
bin: [
  "main.native" { "vhd-tool" }
  "sparse_dd.native" { "sparse_dd" }
]
man: [
  "vhd-tool.1" { "vhd-tool.1" }
  "sparse_dd.1" { "sparse_dd.1" }
]
```

**Comment by gaborigloi** [comment]
It seems that in `vhd-tool`, we're not relying on OASIS to install stuff, but generate a `vhd-tool.install` file with `configure.ml`.

**Comment by gaborigloi** [comment]
It seems that the conditionals in the `Makefile` of `vhd-tool` are wrong:
`ifeq ($(ENABLE_XENSERVER), "--enable-xenserver")` should be `ifeq ($(ENABLE_XENSERVER),--enable-xenserver)`.

And that `sparse_dd` cannot generate `groff` help anymore:
```
ln -s sparse_dd.native sparse_dd
./sparse_dd --help=groff > sparse_dd.1
./sparse_dd: unknown option '--help=groff'.
...
```

**Comment by gaborigloi** [comment]
Closing temporarily, as it seems that the `Makefile` of `vhd-tool` needs to be fixed first, will reopen when we've fixed `vhd-tool`.

**Comment by gaborigloi** [comment]
Opened https://github.com/xapi-project/vhd-tool/pull/41

**Comment by mseri** [comment]
I am re-running the tests now that the makefile has been fixed

**Comment by gaborigloi** [comment]
The previous PR wasn't completely correct: https://github.com/xapi-project/vhd-tool/pull/42

**Comment by gaborigloi** [comment]
Installation, uninstallation, and reinstallation of `vhd-tool` worked for me with the above change in a container, so this PR can be merged. Once the above PR for `vhd-tool` is merged, it will be installable in `xs-opam`.

**Comment by gaborigloi** [comment]
Yay, the build is green! :smile:

**Comment by mseri** [comment]
We need to set `cstruct < '3.0.0'` on all our packages to fix the build against the upstream opam. This is independent of this PR, that can meanwhile be merged

**Pull request merged**